node/actors/network/src/debug_page/mod.rs

```diff
@@ -217,115 +220,448 @@ impl Server {
                     .with_cell(
                         TableCell::new(TableCellType::Header)
-                            .with_attributes([("colspan", "2")])
-                            .with_raw("received [B]"),
-                    )
+                            .with_attributes([("colspan", "3")])
+                            .with_raw("Network address"),
+                    ),
+            )
+            .with_header_row(vec!["", "Address", "Version", "Timestamp"]);
+
+        for (key, addr) in connections {
+            table.add_body_row(vec![
+                key.encode(),
+                addr.msg.addr.to_string(),
+                addr.msg.version.to_string(),
+                addr.msg.timestamp.to_string(),
+            ])
+        }
+
+        table.to_html_string()
+    }
+
+    fn attester_committee_table<'a>(
+        attesters: impl Iterator<Item = &'a attester::WeightedAttester>,
+    ) -> String {
+        let mut table = Table::new().with_header_row(vec!["Public key", "Weight"]);
+
+        for attester in attesters {
+            table.add_body_row(vec![attester.key.encode(), attester.weight.to_string()]);
+        }
+
+        table.to_html_string()
+    }
+
+    fn attester_votes_table<'a>(
+        votes: impl Iterator<
+            Item = (
+                &'a attester::PublicKey,
+                &'a Arc<attester::Signed<attester::Batch>>,
+            ),
+        >,
+    ) -> String {
+        let mut table = Table::new()
+            .with_custom_header_row(
+                TableRow::new()
+                    .with_cell(TableCell::new(TableCellType::Header).with_raw("Public key"))
                     .with_cell(
                         TableCell::new(TableCellType::Header)
-                            .with_attributes([("colspan", "2")])
-                            .with_raw("sent [B]"),
-                    )
-                    .with_cell(TableCell::new(TableCellType::Header).with_raw("Age")),
+                            .with_attributes([("colspan", "3")])
+                            .with_raw("Batch"),
+                    ),
             )
-            .with_header_row(vec!["", "", "", "total", "avg", "total", "avg", ""]);
-        for (key, stats, build_version) in connections {
-            let age = SystemTime::now()
-                .duration_since(stats.established)
-                .ok()
-                .unwrap_or_else(|| Duration::new(1, 0))
-                .max(Duration::new(1, 0)); // Ensure Duration is not 0 to prevent division by zero
-            let received = stats.received.load(Ordering::Relaxed);
-            let sent = stats.sent.load(Ordering::Relaxed);
+            .with_header_row(vec!["", "Number", "Hash", "Genesis hash"]);
+
+        for (key, batch) in votes {
+            table.add_body_row(vec![
+                key.encode(),
+                batch.msg.number.to_string(),
+                batch.msg.hash.encode(),
+                batch.msg.genesis.encode(),
```

Comment:
genesis hash and msg hash are always the same for each vote, so they would obfuscate the displayed information. You can display the data about the message that we are collecting votes for separately.

also IMO it would be more informative to display the committee and votes tables together: row for attester + if we have their vote, so that it is clearly visible who didn't sent their vote yet.

Reply:
Is it guaranteed that it is always the same for all votes? Because the data structure doesn't seem to enforce that.

Ok on the tables though, I can merge.